### PR TITLE
Adding is:exactdupe and is:exactdupelower

### LINKED ISSUE #2482: ghost dupe by perk, not name
The 'is:dupe' should consider each Ghost's perks, not names, when detecting duplicates. DIM obviously knows the perks, so it'd be great if the is:dupe flag worked differently on ghosts. (Or if there was another different flag to use, if that's too complicated.)

re this reddit post: https://redd.it/7bg8x9

### PR DESCRIPTION
**Pull request by kyleshay**
Added two new filters:

`is:exactdupe` will look for items that are the same and have the same
perks

`is:exactdupelower` will look for items that are the same and have the
same perks, but will only highlight items that are less powerful

Can be convinced to make the filter `is:dupeexact` `is:dupelowerexact`
so it shows up in auto complete (or some other name all together)

I think that this resolves #2482

### CODE REVIEW

**Comment by SunburnedGoose** [comment]
I'm fine with it, but I also feel like we've got too many variations on dupe now.

**Comment by kyleshay** [comment]
Agree, maybe we should make a decision about what the most important/happiest paths are

**Comment by kyleshay** [comment]
I also think it would be cool to remove most of the duplication filters, and bring the feature forward as a way for users to effortlessly clear junk from their vaults. 

As an idea, we could have a mode where users could get a popup with details about what items could be dismantled, along with a button to transfer to active character, and have a way for them to revert back to their state before they had dismantled things. (temporary loadout, etc.)

Otherwise, maybe adding a thing next to the collect postmaster button that shows the number of duplicate items, and when clicked, transfers them to that character.

**Comment by SunburnedGoose** [comment]
I would like to have a Cleanup feature. The Janitor.

**Comment by bhollis** [comment]
Agreed on moving this into a higher-order feature. But I'm OK with adding the searches too.

**Comment by duckmanBR** [comment]
I was going to say that this replaces #2737 but then I saw that you begin getting hash duplicates when my idea was a way to check for any items with same perks.
But since I don't know when I'll be able to be back coding, I'll probably close that PR (since everything changed/is going to change).

File: src/app/search/search-filters.ts [hunk 1]
```diff
@@ -542,33 +540,74 @@ export function searchFilters(
...
-              for (const dupe of dupes) {
-                if (dupe.bucket && (dupe.bucket.sort === 'Weapons' || dupe.bucket.sort === 'Armor') && !dupe.notransfer) {
-                  _lowerDupes[dupe.id] = dupe !== bestDupe;
-                }
-              }
+          const groupedDuplicates = _.groupBy(storeService.getAllItems(), (i) => i.hash);
+
+          // reduce the duplicates
+          _duplicates = {};
+          for(let itemHash in groupedDuplicates) {
```
[the remark this answers is not in the document]
Comment (kyleshay) [reply src/app/search/search-filters.ts]: Hmm, thanks! I'll run tslint over my code and resubmit.

File: src/app/search/search-filters.ts [hunk 2]
```diff
@@ -542,33 +540,74 @@ export function searchFilters(
...
-          });
+          }
+        }
+
+        function getPerkList(item) {
+          return item.sockets && item.sockets.sockets.map((socket) => {
+            if (socket.plugOptions.length > 1) {
+                return socket.plugOptions.map((i) => i.plugItem.hash).sort();
+            }
+          }).sort().flat().filter(v=>v);
```
[the remark this answers is not in the document]
Comment (kyleshay) [reply src/app/search/search-filters.ts]: Whoops, good catch. I'll be sure to remove that. Isn't it cool though!

**Comment by kyleshay** [comment]
I'm actually going to close this. I was originally trying to get at a way to compare items that had the same perks, because I knew that most items with the same perks also can be recovered from your collection (items with random rolls don't show up there.) The likelihood of having an exact duplicate of random rolls is pretty low. So for now I'm using is:dupelower is:randomroll and that works great.

**Comment by DHager** [comment]
@kyleshay Thanks for the tip, `not:random is:dupelower` is something I was looking for but didn't know was available from the UI autocompletion hints.

**Pull request closed without merging**